src/main/java/org/elasticsearch/index/engine/internal/InternalEngine.java

```diff
@@ -1135,37 +1150,32 @@ public int compare(Segment o1, Segment o2) {
 
     @Override
     public void close() throws ElasticsearchException {
-        logger.debug("close now acquire writeLock");
-        try (InternalLock _ = writeLock.acquire()) {
```

Comment:
agreed - I think we can just bypass the write lock if we are closing doe to a tragic event instead? I think indexing should fail in such a case no?